### PR TITLE
Update and handle last_modified timestamps correctly.

### PR DESCRIPTION
**Pull request by Kapin**
Depends on #187 because of UI.

Does two things.

1. Fixes updating last_modified timestamps for models on the server so that child elements (pages, elements) cascade their updates to their parents.
2. Makes it so that procedures use their actual last_modified timestamp and are sorted in the correct order.

### CODE REVIEW

File: src-backend/api/tests/test_model_procedure.py [hunk 1]
```diff
@@ -71,3 +72,13 @@ def test_owner_none(self):
...
+
+    def test_updates_last_modified(self):
+        procedure = factories.ProcedureFactory()
+        original_last_modified = procedure.last_modified
+
+        factories.PageFactory(
+            procedure=procedure
+        )
+
+        assert_not_equals(original_last_modified, procedure.last_modified)
```
Comment (Trinovantes) [inline src-backend/api/tests/test_model_procedure.py]: should assert current `last_modified` is greater than `original_last_modified`

File: src-backend/api/tests/test_model_page.py [hunk 2]
```diff
@@ -64,3 +65,13 @@ def test_display_index_uniqueness(self):
...
+
+    def test_updates_last_modified(self):
+        page = factories.PageFactory()
+        original_last_modified = page.last_modified
+
+        factories.ElementFactory(
+            page=page
+        )
+
+        assert_not_equals(original_last_modified, page.last_modified)
```
Comment (Trinovantes) [inline src-backend/api/tests/test_model_page.py]: here too

**Comment by Trinovantes** [comment]
shouldn't modifying elements/choices change last modified too?

File: src-backend/api/tests/test_model_procedure.py (hunk 1 above)
Comment (Kapin) [inline src-backend/api/tests/test_model_procedure.py]: will try

File: src-backend/api/tests/test_model_page.py (hunk 2 above)
Comment (Kapin) [inline src-backend/api/tests/test_model_page.py]: see above

**Comment by Kapin** [comment]
@Trinovantes it should cascade, but I'll verify.

File: src-frontend/app/controllers/procedures/index.js [hunk 3]
```diff
@@ -1,6 +1,10 @@
 import Ember from 'ember';
 
 export default Ember.Controller.extend({
+    procedures: function() {
+        return this.get('model').sortBy('lastModified').reverse();
```
Comment (Ommy) [inline src-frontend/app/controllers/procedures/index.js]: Wish there was a cleaner way to do this.

Comment (Kapin) [inline src-frontend/app/controllers/procedures/index.js]: @ConnorCimowsky probably knows a better way

File: src-frontend/app/helpers/date-string.js [hunk 4]
```diff
@@ -0,0 +1,7 @@
+import Ember from 'ember';
```
Comment (Ommy) [inline src-frontend/app/helpers/date-string.js]: Are we really going to have a new file for every handlebar helper function?

Comment (ConnorCimowsky) [inline src-frontend/app/helpers/date-string.js]: Yup, this is the convention

File: src-frontend/app/controllers/procedures/index.js (hunk 3 above)
Comment (ConnorCimowsky) [inline src-frontend/app/controllers/procedures/index.js]: Would this work?

```javascript
procedures: function() {
  return Ember.ArrayProxy.createWithMixins(Ember.SortableMixin, {
    sortProperties: ['lastModified'],
    sortAscending: false,
    content: this.get('model')
  });
}.property('model.@each.lastModified')
```

**Comment by ConnorCimowsky** [comment]
LGTM after comments

**Comment by Ommy** [comment]
+1 after comments

File: src-frontend/app/controllers/procedures/index.js (hunk 3 above)
Comment (Kapin) [inline src-frontend/app/controllers/procedures/index.js]: I'll try that

**Comment by Trinovantes** [comment]
+1

**Pull request merged**